tests/test_testclient.py

```diff
@@ -273,3 +274,48 @@ def homepage(request):
     client = test_client_factory(app)
     response = client.get("/", params={"param": param})
     assert response.text == param
+
+
+@pytest.mark.parametrize(
+    "domain, ok",
+    [
+        pytest.param(
+            "testserver",
+            True,
+            marks=[
+                pytest.mark.xfail(
+                    sys.version_info < (3, 11),
+                    reason="Fails due to domain handling in http.cookiejar module (see "
+                    "#2152)",
+                ),
+            ],
+        ),
+        ("testserver.local", True),
+        ("localhost", False),
+        ("example.com", False),
+    ],
+)
+def test_domain_restricted_cookies(test_client_factory, monkeypatch, domain, ok):
```

Comment:
Why is the `monkeypatch` here?
```suggestion
def test_domain_restricted_cookies(test_client_factory, domain, ok):
```

Reply:
Seems to be left over somehow.